da/celestia/celestia.go

```diff
@@ -399,6 +400,17 @@ func (c *DataAvailabilityLayerClient) retrieveBatchesNoCommitment(dataLayerHeigh
 
 func (c *DataAvailabilityLayerClient) CheckBatchAvailability(daMetaData *da.DASubmitMetaData) da.ResultCheckBatch {
 	var availabilityResult da.ResultCheckBatch
+	//check namespace
+	if !slices.Equal(daMetaData.Namespace, c.config.NamespaceID.Bytes()) {
```

Comment:
why isn't this check one on the `fetchBatch`? also may be worth to simply write a `validateBasic` method on the daMetaData after fetching and test it all matches the current configuration